pom.xml

```diff
@@ -751,6 +747,171 @@
           </formats>
         </configuration>
       </plugin>
+
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
+        <version>${maven.jacoco.plugin.version}</version>
+        <executions>
+          <execution>
+            <id>default-instrument</id>
+            <goals>
+              <goal>instrument</goal>
+            </goals>
+            <phase>test-compile</phase>
+          </execution>
+          <execution>
+            <id>default-restore-instrumented-classes</id>
+            <goals>
+              <goal>restore-instrumented-classes</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>default-report</id>
+            <phase>site</phase>
+            <goals>
+              <goal>report</goal>
```

Comment:
please do:
```
$ git diff
diff --git a/pom.xml b/pom.xml
index dda176c..b83192e 100644
--- a/pom.xml
+++ b/pom.xml
@@ -768,10 +768,13 @@
           </execution>
           <execution>
             <id>default-report</id>
-            <phase>site</phase>
+            <phase>verify</phase>
             <goals>
               <goal>report</goal>
             </goals>
+            <configuration>
+              <dataFile>${project.build.directory}/jacoco/jacoco.exec</dataFile>
+            </configuration>
```

it is better to generate report on verify phase, to have report available right after verification is failed. We ask user to use "mvn verify" for all verifications. Without html report it is impossible to get to know what to fix. All other plugins report to terminal output some report that is enough to fix problems.

Reply:
done

Reply:
reverted back to the `site` phase since `verify` is no more required to get the JaCoCo report.